x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -239,14 +239,21 @@ export class CsvGenerator {
   }
 
   public async generateData(): Promise<TaskRunResult> {
+    const searchSourceFields = {
+      ...(this.job.searchSource ?? {}),
```

Comment:
it looks to me that the passed in searchsource should be configured correctly. When we start using csv exports in other applications they might not want this same behavior discover uses. That's why discover should be passing in fully configured search source imo.

cc @vadimkibana